File: source/c/bst_levelorder.c
// gcc -o bst_levelorder bst_levelorder.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node {
    int value;
    struct Node *left, *right;
} Node;

Node *new_node(int value) {
    Node *n = malloc(sizeof(Node));
    n->value = value; n->left = n->right = NULL;
    return n;
}

Node *insert(Node *root, int value) {
    if (!root) return new_node(value);
    if (value < root->value) root->left = insert(root->left, value);
    else if (value > root->value) root->right = insert(root->right, value);
    return root;
}

int main(void) {
    int values[] = {50, 30, 70, 20, 40, 60, 80};
    Node *root = NULL;
    for (int i = 0; i < 7; i++) root = insert(root, values[i]);

    // BFS with a simple queue
    Node *queue[100];
    int head = 0, tail = 0;
    int result[100];
    int result_len = 0;

    if (root) queue[tail++] = root;
    while (head < tail) {
        Node *node = queue[head++];
        result[result_len++] = node->value;
        if (node->left) queue[tail++] = node->left;
        if (node->right) queue[tail++] = node->right;
    }

    printf("levelorder:");
    for (int i = 0; i < result_len; i++) printf(" %d", result[i]);
    printf("\n");
    return 0;
}

File: source/javascript/bst_levelorder.js
// BST level-order traversal (BFS using array as queue)
function newNode(value) { return { value, left: null, right: null }; }

function insert(root, value) {
    if (root === null) return newNode(value);
    if (value < root.value) root.left = insert(root.left, value);
    else if (value > root.value) root.right = insert(root.right, value);
    return root;
}

function levelorder(root) {
    if (root === null) return [];
    const result = [];
    const queue = [root];
    while (queue.length > 0) {
        const node = queue.shift();
        result.push(node.value);
        if (node.left !== null) queue.push(node.left);
        if (node.right !== null) queue.push(node.right);
    }
    return result;
}

let root = null;
const values = [50, 30, 70, 20, 40, 60, 80];
for (const v of values) root = insert(root, v);

const result = levelorder(root);
console.log(`levelorder: ${result.join(" ")}`);
